src/UI/Actions.hs

```diff
@@ -844,9 +845,10 @@ updateBrowseFileContents contents s =
 applySearch :: (MonadIO m) => AppState -> m AppState
 applySearch s = do
   r <- runExceptT (Notmuch.getThreads searchterms (view (asConfig . confNotmuch) s))
+  t <- liftIO (zonedTimeToUTC <$> getZonedTime)
   case r of
     Left e -> pure $ setError e s
-    Right threads -> updateList threads <$> notifyNumThreads s threads
+    Right threads -> updateList threads <$> notifyNumThreads s threads >>= pure . set asLocalTime t
```

Comment:
```haskell
Right threads -> set asLocalTime t . updateList threads <$> notifyNumThreads s threads
```

`(>>= pure . f)` can always be replaced with a simple `fmap f` :)

Reply:
Done. Had some trouble with the `fmap` but hopefully got it right.